app/src/main/java/com/example/calculator/MainActivity.java

```diff
@@ -23,25 +23,42 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     private void setListeners() {
-        setUpClickListenerToButton(binding.buttonAdd);
-        setUpClickListenerToButton(binding.buttonSub);
-        setUpClickListenerToButton(binding.buttonProduct);
-        setUpClickListenerToButton(binding.buttonDivision);
-        setUpClickListenerToButton(binding.buttonEquals);
-        setUpClickListenerToButton(binding.buttonClear);
-        setUpClickListenerToButton(binding.buttonZero);
-        setUpClickListenerToButton(binding.buttonOne);
-        setUpClickListenerToButton(binding.buttonTwo);
-        setUpClickListenerToButton(binding.buttonThree);
-        setUpClickListenerToButton(binding.buttonFour);
-        setUpClickListenerToButton(binding.buttonFive);
-        setUpClickListenerToButton(binding.buttonSix);
-        setUpClickListenerToButton(binding.buttonSeven);
-        setUpClickListenerToButton(binding.buttonEight);
-        setUpClickListenerToButton(binding.buttonNine);
+        setUpClickListenerToOperatorButton(binding.buttonAdd);
+        setUpClickListenerToOperatorButton(binding.buttonSub);
+        setUpClickListenerToOperatorButton(binding.buttonProduct);
+        setUpClickListenerToOperatorButton(binding.buttonDivision);
+        setUpClickListenerToEqualsButton(binding.buttonEquals);
+        setUpClickListenerToClearLastButton(binding.buttonClear);
+        setUpClickListenerToClearAllButton(binding.buttonClear);
+        setUpClickListenerToNumberButton(binding.buttonZero);
+        setUpClickListenerToNumberButton(binding.buttonOne);
+        setUpClickListenerToNumberButton(binding.buttonTwo);
+        setUpClickListenerToNumberButton(binding.buttonThree);
+        setUpClickListenerToNumberButton(binding.buttonFour);
+        setUpClickListenerToNumberButton(binding.buttonFive);
+        setUpClickListenerToNumberButton(binding.buttonSix);
+        setUpClickListenerToNumberButton(binding.buttonSeven);
+        setUpClickListenerToNumberButton(binding.buttonEight);
+        setUpClickListenerToNumberButton(binding.buttonNine);
     }
 
-    private void setUpClickListenerToButton(Button buttonPressed) {
-        buttonPressed.setOnClickListener(view -> presenter.onButtonPressed(buttonPressed.getText().toString()));
+    private void setUpClickListenerToNumberButton(Button buttonPressed) {
```

Comment:
if this method is only called one time, I think is not necessary. You can call directly in the listener

Reply:
Solved!